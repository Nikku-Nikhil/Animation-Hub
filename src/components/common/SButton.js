import {StyleSheet, Text, TouchableOpacity} from 'react-native';
import React from 'react';
import SText from './SText';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';

const SButton = ({
  text,
  textSize = 14,
  backgroundColor = '#eee',
  color = '#000',
  width = '100%',
  containerStyle,
  iconShown = true,
  borderColor = '#000',
  icon = <MaterialIcons name="arrow-forward-ios" size={20} />,
  onPress = () => {},
}) => {
  console.log("Text",text);
  return (
    <TouchableOpacity
      onPress={onPress}
      style={[
        {
          backgroundColor,
          width,
          borderColor,
        },
        styles.btn,
        containerStyle,
      ]}>
      <SText text={text} color={color} size={textSize} />
      {iconShown && icon}
    </TouchableOpacity>
  );
};

export default SButton;

const styles = StyleSheet.create({
  btn: {
    padding: 18,
    justifyContent: 'space-between',
    flexDirection: 'row',
    borderRadius: 8,
    borderWidth: 1,
    alignItems: 'center',
    marginVertical: 5,
  },
});
