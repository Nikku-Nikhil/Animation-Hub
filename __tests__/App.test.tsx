import 'react-native';
import React from 'react';
import SButton from '../src/components/common/SButton';

import renderer from 'react-test-renderer';


  test('App test renderer snapshot', () => {
    const snapshot = renderer.create(<SButton />).toJSON();
    console.log(snapshot);
    expect(snapshot).toMatchSnapshot();
  });
