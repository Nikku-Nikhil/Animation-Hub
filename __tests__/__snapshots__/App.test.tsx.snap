// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` App test renderer snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#eee",
      "borderColor": "#000",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 5,
      "opacity": 1,
      "padding": 18,
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
          "fontSize": 14,
          "fontWeight": 500,
          "lineHeight": undefined,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`App test renderer snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#eee",
      "borderColor": "#000",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 5,
      "opacity": 1,
      "padding": 18,
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
          "fontSize": 14,
          "fontWeight": 500,
          "lineHeight": undefined,
        },
        undefined,
      ]
    }
  />
</View>
`;
